### DISCUSSION TITLE
FYI: Outlook doesn't handle full-day events correctly

### DISCUSSION THREAD

**Discussion by Kaifeck** [spatie/icalendar-generator#94] [Q&A] [answered]
We are using this library and recently encountered a bug where an Outlook user reported error messages when importing a calendar (with all dates being full-day).
After testing locally, the error turned out to be:
> Task 'Internet Calendar' reported error (0x0004003D) : 'The VEVENT, "Event", defined near line xyz, lacks a DTEND. Outlook is assuming that this appointment is an hour long. Double-click to open this item.'

After checking the .ics file we found that it did lack DTENDs. However, according to the [RFC](https://www.rfc-editor.org/rfc/rfc5545#page-54) this is allowed:
>  For cases where a "VEVENT" calendar component
      specifies a "DTSTART" property with a DATE value type but no
      "DTEND" nor "DURATION" property, the event's duration is taken to
      be one day.

So it turned out that Outlook does not implement the RFC Specs correctly. After some searching this seems to have been a Problem for [quite a while](https://answers.microsoft.com/en-us/outlook_com/forum/all/bug-outlook-2013-throws-error-when-icalendar-is/fa136faf-f410-4c08-bfc2-85248723c27d).

This isn't a bug report. Just a heads up in case someone else runs into the same Problem. Our fix was that we added an enddate to the Event.

**Comment by rubenvanassche** [comment, marked as answer]
Thanks for the information @Kaifeck! Maybe we should add a file to the repo containing all sorts of oddities in calendar clients since this is not the first one and it will probably not be the last one 😛
